### ISSUE TITLE
SkyWeave insane FPS drops with certain blocks

### ISSUE THREAD

**Issue by ArtichaTM** [baileyholl/Ars-Nouveau#1221]
Using many skyweave is ok, but dispelling them (converting to simple mirrorweave) drops FPS quite drastically. Funny part is, some blocks working "OK", dealing not so much damage to framerate, but some blocks forces FPS to jump from 77 to 19. [Example](https://youtu.be/crx2B7KOjp4)
Causes heavy lags (drops from ~77 to ~19): dirt, planks (Any, vanilla/mods), awakened supremium gemstone blocks
Causes non-heavy lags (~77 to ~50): crying obsidian

**Comment by baileyholl** [comment]
I am curious if Entity Culling improves this for you - MC does not cull block entity rendering like it does for normal placed blocks. It will always be more laggy than a normal static block with no model though. Thats why facade mods go out of their way to register extra blockstates for every texture in the game, rather than one pretend block.

**Comment by ArtichaTM** [comment]
> I am curious if Entity Culling improves this for you - MC does not cull block entity rendering like it does for normal placed blocks. It will always be more laggy than a normal static block with no model though. Thats why facade mods go out of their way to register extra blockstates for every texture in the game, rather than one pretend block.

No, I have no problem with lags causing by SkyWeave, I can't understand why certain static blocks causing insane fps drops. Crying obsidian, block with some animation does not drop fps, but dirt, DIRT causes insane fps drop by 300-400%. I just want to know, what the difference between this 2 blocks that causes this absurd difference in FPS

**Comment by baileyholl** [comment]
I understand what you are saying. They use different forms of rendering, Skyweave applies a baked shader like the skybox and has less work to do. If you add entity culling I am curious if it improves FPS for mirrorweave and camoflauged skyweave.
